torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2537,6 +2567,57 @@ void ProcessGroupNCCL::broadcastUniqueNCCLID(
   }
 }
 
+void ProcessGroupNCCL::allgatherUniqueNCCLID(
+    int rootRank,
+    ncclUniqueId* ncclID,
+    std::vector<ncclUniqueId>& ncclIDs) {
+  std::vector<std::string> storeKeys;
+  for (size_t r = 0; r < ncclIDs.size(); r++) {
+    storeKeys.emplace_back("UniqueNCCLID:" + std::to_string(r));
+  }
+  if (rank_ == rootRank) {
+    auto vec = std::vector<uint8_t>(
+        reinterpret_cast<uint8_t*>(ncclID),
+        reinterpret_cast<uint8_t*>(ncclID) + NCCL_UNIQUE_ID_BYTES);
+    store_->set(storeKeys[rootRank], vec);
+  }
```

Comment:
I wonder if `storeKeys[rootRank]` would overflow, because `storeKeys` has length of `numRoots`, but `if (rank_ == rootRank)` seems to suggest that `rootRank` can range from 0 to `world_size - 1`.

Reply:
technically yes, but in reality, we use 128 root per rank, do you still think it is possible? Or I can add a warning there?

Reply:
I don't know what `rootRank` stands for, is my point.

Reply:
it actually should be rootIdx..